woocommerce/woocommerce
# Use simple-git to produce diffs in the code analyzer

### All Submissions:






### Changes proposed in this Pull Request:

As part of work to fill the beta/release post, I needed a simpler way to generate diffs that didn't involve working on your local copy of WooCommerce. Whats nice about that is that any uncommitted changes on your local env won't break the analyzer like it does now.

This starts the work to complete that by cloning the repo locally in a tmp folder and performing git operations on the copy instead of your local files. To do this I migrated from `execSync` bash scripts to using [simple-git](https://www.npmjs.com/package/simple-git).

Simple git provides a few benefits:

1. Quite often has better error handling than straight output from commands
2. TS support 
3. Greatly simplifies performing relatively complex sets of git operations
4. It makes the code so much more flexible, allowing us to operate quite easily on virtually any local or remote git repository with ease.

I made the decision not to migrate everything to simple-git in one go, but to iterate with smaller PRs. This replaces the diff scanning functionality with simple-git, but leaves the DB schema scanning code in place for now, which still uses the old `execSync` commands. It is my intent to replace the DB schema scanning as a follow up.

Note that we won't get the full benefit of operating on a copy of the files until we finish migrating the DB scanning code too.



### How to test the changes in this Pull Request:

1. It would help if you know a previous command run that produces some interesting output of hook and template changes
2. Input that command slightly changed for the command option changes (from root of project): `./tools/code-analyzer/bin/dev analyzer "<YOUR COMMIT HASH TO COMPARE>"`
3. See that the output of hook and template changes looks reasonable.

### Other information:




### FOR PR REVIEWER ONLY:
